src/d3a/d3a_core/simulation.py

```diff
@@ -290,6 +290,12 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
                     (tick_no + 1) / config.ticks_per_slot * 100,
                 )
 
+                if ConstSettings.GeneralSettings.DISPATCH_EVENTS_BOTTOM_TO_TOP:
+                    # trigger market_cycle:
+                    if self.area.current_tick % self.area.config.ticks_per_slot == 0:
```

Comment:
This check could be avoided and this code block could be moved out of this for loop too.

Reply:
True that! Will try it.